app/view_objects/apply_invalid_data_view.rb

```diff
@@ -4,49 +4,81 @@ class ApplyInvalidDataView
   include ActionView::Helpers::UrlHelper
   include ActionView::Helpers::TagHelper
 
-  def initialize(apply_application)
+  APPLY_INVALID_DATA_FIELDS_ORDER = %w[
+    institution
+    country
+    subject
+    uk_degree
+    non_uk_degree
+    grade
+    graduation_year
+  ].freeze
+
+  def initialize(apply_application, degree: nil, on_form_page: false)
     @apply_application = apply_application
+    @degree = degree
+    @on_form_page = on_form_page
     @invalid_fields = populate_invalid_fields
   end
 
   def summary_content
-    I18n.t("views.apply_invalid_data_view.invalid_answers_summary", count: invalid_fields.size)
+    I18n.t("views.apply_invalid_data_view.invalid_answers_summary", count: invalid_fields.flatten.size)
   end
 
   def summary_items_content
     @summary_items_content ||= safe_join(
-      invalid_fields.map do |field|
-        tag.li(
-          link_to(
-            I18n.t("views.apply_invalid_data_view.unrecognised_field_text", invalid_field: field.to_s.upcase_first),
-            "##{field}",
-            class: "govuk-notification-banner__link",
-          ),
-        )
+      invalid_fields.map.with_index(1) do |fieldset, index|
+        fieldset.map do |field|
+          tag.li(
+            link_to(
+              I18n.t("views.apply_invalid_data_view.unrecognised_field_text", invalid_field: field.to_s.humanize.upcase_first),
+              get_link_anchor(field, index),
+              class: "govuk-notification-banner__link",
+            ),
+          )
+        end
       end,
     )
   end
 
   def invalid_data?
-    invalid_fields.size.positive?
+    invalid_fields.flatten.size.positive?
   end
 
 private
 
-  attr_reader :apply_application, :invalid_fields
+  attr_reader :apply_application, :invalid_fields, :degree
+
+  def get_link_anchor(field, index)
+    return get_form_page_link_anchor(field) if @on_form_page
+    return "##{field.parameterize}" if invalid_fields.size == 1
+
+    "##{field.parameterize}-#{index}"
+  end
+
+  def get_form_page_link_anchor(field)
+    "#degree-#{field.parameterize}-field" if invalid_fields.size == 1
```

Comment:
```suggestion
    "#degree-#{field.parameterize}-field"
```
Do we need the condition here at all?